dodola/tests/test_services.py

```diff
@@ -84,6 +91,114 @@ def domain_file(request):
     return domain
 
 
+def test_apply_qdm(tmpdir):
+    """Test to apply a trained QDM to input data and read the output.
+
+    This is an integration test between train_qdm, apply_qdm.
+    """
+    # Setup input data.
+    target_variable = "fakevariable"
+    variable_kind = "temperature"
+    n_histdays = 10 * 365  # 10 years of daily historical.
+    n_simdays = 50 * 365  # 50 years of daily simulation.
+
+    model_bias = 2
+    ts_ref = np.ones(n_histdays, dtype=np.float64)
+    ts_sim = np.ones(n_simdays, dtype=np.float64)
+    hist = _datafactory(ts_ref + model_bias, variable_name=target_variable)
+    ref = _datafactory(ts_ref, variable_name=target_variable)
+    sim = _datafactory(ts_sim + model_bias, variable_name=target_variable)
+
+    # Load up a fake repo with our input data in the place of big data and cloud
+    # storage.
+    qdm_key = "memory://test_apply_qdm/qdm.zarr"
+    hist_key = "memory://test_apply_qdm/hist.zarr"
+    ref_key = "memory://test_apply_qdm/ref.zarr"
+    sim_key = "memory://test_apply_qdm/sim.zarr"
+    # Writes NC to local disk, so dif format here:
+    sim_adjusted_key = tmpdir.join("sim_adjusted.nc")
+    repository.write(sim_key, sim)
+    repository.write(hist_key, hist)
+    repository.write(ref_key, ref)
+
+    # Target the earliest year we can:
+    target_year, _ = qdm_rollingyearwindow(sim)
+
+    train_qdm(
+        historical=hist_key,
+        reference=ref_key,
+        out=qdm_key,
+        variable=target_variable,
+        kind=variable_kind,
+    )
+    apply_qdm(
+        simulation=sim_key,
+        qdm=qdm_key,
+        year=target_year,
+        variable=target_variable,
+        out=sim_adjusted_key,
+    )
+    adjusted_ds = xr.open_dataset(str(sim_adjusted_key))
+    assert target_variable in adjusted_ds.variables
+
+
+@pytest.mark.parametrize("kind", ["precipitation", "temperature"])
+def test_train_qdm(kind):
+    """Test that train_qdm outputs store giving sdba.adjustment.QuantileDeltaMapping
+
+    Checks that output is consistent if we do "temperature" or "precipitation"
+    QDM kinds.
```

Comment:
update this to be additive or multiplicative kinds

Reply:
Cool. I got this in `fbe8d03f`. Marking resolved.